### ISSUE TITLE
Weird code merging from deepseek

### ISSUE THREAD

**Issue by viraptor** [plandex-ai/plandex#184]
I've tried using plandex with deepseek. The responses I see in `plandex convo` seem completely fine and include the whole file being changed.
Unfortunately plandex is spinning for a very long time processing those answers and finally injects a lot of copied code - the new code gets pasted in the middle of the file being edited. (python code)

I'd love for those edits to be handled better.

**Comment by viraptor** [comment]
Aider seems to ask for a search/replace pair when working with deepseek and for any issues applying it it just complains to the LLM and makes it try again. Seems to work quite well in practice.

**Comment by danenania** [comment]
Are you using deepseek for all the [https://docs.plandex.ai/models/roles](roles)? You can check with `plandex models`.

There's a syntax check and a verification step as well, but it may be that deepseek is having trouble with Plandex's line-number based editing. There are pros and cons to this approach vs. aider's search/replace pair strategy... the solution will likely be to change strategies depending on which model is being used, which we are working towards. The prerequisite is a good set of evals that lets us compare different prompting/editing strategies against different models, and we are close to having this in place. So stay tuned :)

**Comment by viraptor** [comment]
Yup, all the roles. It seems like everything works fine, apart from the change merging.

For the editing format, yes, aider uses diff instead for deepseek coder. They already did that homework here so it can save you some time 😉 https://aider.chat/docs/leaderboards/ (includes % reliability for the best method)

**Comment by danenania** [comment]
There's a more robust system for builds/file edits in 2.0.0 that should work better with all kinds of models.
